compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/diagnostic/BLangDiagnosticLocation.java

```diff
@@ -34,12 +34,18 @@
     private LineRange lineRange;
     private TextRange textRange;
 
+    @Deprecated
     public BLangDiagnosticLocation(String filePath, int startLine, int endLine, int startColumn, int endColumn) {
         this.lineRange = LineRange.from(filePath, LinePosition.from(startLine, startColumn),
                 LinePosition.from(endLine, endColumn));
         this.textRange = TextRange.from(0, 0);
     }
 
+    public BLangDiagnosticLocation(LineRange lineRange, TextRange textRange) {
```

Comment:
Looks like this constructor is never used

Reply:
Yeah, this was asked to introduce

Reply:
Since these are internal classes, we can add it when we have a solid a requirement

Reply:
okay